src/service/FileTransferService.ts

```diff
@@ -42,10 +47,62 @@ export class FileTransferService {
                 }
             }).catch((err) => {
                 if (err.code === UNSUPPORTED_MEDIA_TYPE) {
-                    throw new Error('Unsupported file type')
+                    throw new Error('Unsupported file type');
                 } else {
-                    throw new Error(err.message)
+                    throw new Error(err.message);
                 }
             });
     }
+
+    async fileMetada(fileId: string): Promise<FileMetada> {
+        const config: AxiosRequestConfig = {
+            headers: {
+                'x-api-key': this.key
+            },
+        };
+
+        return await axios
+            .get<FileMetada>(`${this.url}/${fileId}`, config)
+            .then(_ => _.data)
+            .catch(_ => {
+                throw new FileNotFoundError(fileId, _.message);
```

Comment:
An API should also handle other scenarios e.g. non 5xx response. We would not like service classify 5xx responses as FileNotFoundError. I guess we can have check whether 404 for returned and throw FileNotFoundError and generic error in other cases (fallback).